code/go/0chain.net/smartcontract/minersc/sharder.go

```diff
@@ -149,7 +150,7 @@ func (msc *MinerSmartContract) getShardersList(balances cstate.StateContextI,
 	}
 	err = allMinersList.Decode(allMinersBytes.Encode())
 	if err != nil {
-		return nil, err
+		return nil, smartcontract.NewError(smartcontract.DecodingErr, err)
```

Comment:
The decoding error message seems too generic. Maybe we could add more context into the message as it's being used in API.

Reply:
Done.